tests/wildcard.rs

```diff
@@ -1,225 +1,251 @@
-// }
-
-// #[test]
-// fn invalid_wildcard() {
-//     let mut app = tide::Server::new();
-//     app.at("/echo/*path/:one/").get(echo_path);
-//     let mut server = make_server(app.into_http_service()).unwrap();
-
-//     let req = http::Request::get("/echo/one/two")
-//         .body(Body::empty())
-//         .unwrap();
-//     let res = server.simulate(req).unwrap();
-//     assert_eq!(res.status(), 404);
-// }
-
-// #[test]
-// fn nameless_wildcard() {
-//     let mut app = tide::Server::new();
-//     app.at("/echo/:").get(|_| async move { "" });
-
-//     let mut server = make_server(app.into_http_service()).unwrap();
-
-//     let req = http::Request::get("/echo/one/two")
-//         .body(Body::empty())
-//         .unwrap();
-//     let res = server.simulate(req).unwrap();
-//     assert_eq!(res.status(), 404);
-
-//     let req = http::Request::get("/echo/one").body(Body::empty()).unwrap();
-//     let res = server.simulate(req).unwrap();
-//     assert_eq!(res.status(), 200);
-// }
-
-// #[test]
-// fn nameless_internal_wildcard() {
-//     let mut app = tide::Server::new();
-//     app.at("/echo/:/:path").get(echo_path);
-//     let mut server = make_server(app.into_http_service()).unwrap();
-
-//     let req = http::Request::get("/echo/one").body(Body::empty()).unwrap();
-//     let res = server.simulate(req).unwrap();
-//     assert_eq!(res.status(), 404);
-
-//     let req = http::Request::get("/echo/one/two")
-//         .body(Body::empty())
-//         .unwrap();
-//     let res = server.simulate(req).unwrap();
-//     assert_eq!(res.status(), 200);
-//     let body = block_on(res.into_body().into_vec()).unwrap();
-//     assert_eq!(&*body, &*b"two");
-
-//     let req = http::Request::get("/echo/one/two")
-//         .body(Body::empty())
-//         .unwrap();
-//     let res = server.simulate(req).unwrap();
-//     assert_eq!(res.status(), 200);
-//     let body = block_on(res.into_body().into_vec()).unwrap();
-//     assert_eq!(&*body, &*b"two");
-// }
-
-// #[test]
-// fn nameless_internal_wildcard2() {
-//     let mut app = tide::Server::new();
-//     app.at("/echo/:/:path").get(echo_empty);
-//     let mut server = make_server(app.into_http_service()).unwrap();
-
-//     let req = http::Request::get("/echo/one/two")
-//         .body(Body::empty())
-//         .unwrap();
-//     let res = server.simulate(req).unwrap();
-//     assert_eq!(res.status(), 200);
-//     let body = block_on(res.into_body().into_vec()).unwrap();
-//     assert_eq!(&*body, &*b"one");
-// }
+use http_types::{Method, StatusCode, Url};
+use tide::{http, Request};
+
+async fn add_one(cx: Request<()>) -> Result<String, tide::Error> {
+    return match cx.param::<i64>("num") {
+        Ok(num) => Ok((num + 1).to_string()),
+        Err(err) => Err(tide::Error::new(StatusCode::BadRequest, err)),
+    };
+}
+
+async fn add_two(cx: Request<()>) -> Result<String, tide::Error> {
+    let one;
+    match cx.param::<i64>("one") {
```

Comment:
this might read more idiomatically if `one` and `two` were returned by the match expression instead of assigned inside of the match expression, and the error early-returning could might be accomplished with a `.map_err` and a `?`

Reply:
Thank you for review.
This corrected at af3b75f.